File: 102.Binary_Tree_Level_Order_Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList();
        List<List<Integer>> res = new ArrayList();
        if (root == null) return res;
        
        queue.offer(root);
        while (!queue.isEmpty()) {
            int levelNum = queue.size();
            List<Integer> currLevel = new ArrayList();
            for (int i=0; i<levelNum; i++) {
                if (queue.peek().left != null) queue.offer(queue.peek().left);
                if (queue.peek().right != null) queue.offer(queue.peek().right);
                currLevel.add(queue.poll().val);
            }
            res.add(currLevel);
        }
        return res;
    }
}

File: 102.BinaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        res = []
        queue = []
        queue.append(root)
        while (queue):
            size = len(queue)
            tmp = []
            for i in range(size):
                curr_node = queue.pop(0)
                tmp.append(curr_node.val)
                if (curr_node.left != None):
                    queue.append(curr_node.left)
                if (curr_node.right != None):
                    queue.append(curr_node.right)
            res.append(tmp)
        return res
                
        